connect/src/spirc.rs

```diff
@@ -438,20 +438,15 @@ impl SpircTask {
                         error!("error updating connect state for volume update: {why}")
                     }
                 },
-                else => break
-            }
-        }
-
-        if !self.shutdown && self.connect_state.is_active() {
-            if let Err(why) = self.notify().await {
-                warn!("notify before unexpected shutdown couldn't be send: {why}")
+                else => {
```

Comment:
Btw. I think this is not completly correct. I think i had a wrong idea how this part works, but the disconnect still has to occure outside. Otherwise branches can stil break and by that skip the disconnect here.

Reply:
The general code flow didn't change with the dealer implementation (and this here is the `tokio::select!` main loop in the `SpircTask::run` method that existed before :D). Just the underlying logic changed pretty drastically, but thats because the dealer handles informations and the queue quite different.

Anyhow, thanks for the hint! I will look into it later. Didn't even know of `ControlFlow` yet.